staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -106,7 +106,12 @@ func lookupServiceName(name string) (EgressType, error) {
 }
 
 func tunnelHTTPConnect(proxyConn net.Conn, proxyAddress, addr string) (net.Conn, error) {
```

Comment:
this definitely needs an accompanying test demonstrating the problem and what this is fixing

Reply:
Normally that's my line, but this looks wrong enough that I'm surprised it works...

Reply:
> this looks wrong enough that I'm surprised it works...

maybe it doesn't? in which case we still need something exercising it :)

Reply:
I think the old code accidentally worked. We had even done performance tests for this CONNECT code path.

@Jefftree you mentioned that this worked because of golang internals, do you have link to the code?